### ISSUE TITLE
Change to cell span handle behavior

### ISSUE THREAD

**Issue by ladlon** [turtletooth/tahoma2d#437]
Regarding the cell span handle (...I think it's referred to as the Smart Handle?...  The little tab that appears on the bottom/right of the selected cell)...  The way it behaves when you pull down/right is fine, but I've always wondered (...been annoyed by...) why you can't pull up/left to shrink the span (as you do to expand it, pulling down/right).

Instead, it seems you have to select a few cells before it, and THEN you can pull up/left.

It always seems like unnecessary busy work, and I could never understand why you can't just pull up/left.

Whether it then stops if you hit a drawing key (dot), or allows you to keep going further (effectivly deleting whatever you pass over) is a matter of debate, I suppose... (...I'd allow the latter, as there's nothing stopping you from just manually stopping the drag if you don't want to erase the previous 'keys').

I find myself frequently trying to instinctvely pull up/left, only to find I can't, and then remembering I have to select the keys before it (although, again, I see no reason why that is needed, as the only 'reason' I can see is to set an 'affect range'... which seems pointless, as you can simply move as much as you want, without the need for some pre-determined 'range' to then stop you when you hit its end.

Certainly would make workflow a lot faster/easier... for me at least, since I can't speak for others.  Just seems pointless extra work, with seemingly no purpose or need.  It would also make it consistant to the action of expanding.   (Actually, I find that Tahoma2D seems to have a problem with consistancy in the control schemes... like use of modifiers, etc.... which, in turn, does make it a bit harder to learn/remember the various variations in editing actions.

**Comment by manongjohn** [comment]
I don't have an issue shrinking a held cell range using the smart tab. I don't understand having to go up a few cells to get it to work.

Perhaps I am misunderstanding what you are doing and how you are working around it. Can you provide an animated gif or link to video to show the issue you are having?

**Comment by ladlon** [comment]
Sure (I can do that now!)...

Here is a vid of me selecting the last cell of a span, and first dragging right (works as expected), and then I try dragging left (and it won't go).  I then select a random number of frames before the end of the span, and now it lets me move it.

I'm not understanding why I can't just drag left (just as you can when you move right)... and find the necessity of first having to select some cells before it unneccessary work, and seemingly pointless.

![WorkPlaneNotAligning](https://user-images.githubusercontent.com/71803483/97504533-6efa3680-194d-11eb-9a57-0c3cd68cda5b.gif)

**Comment by manongjohn** [comment]
ok. I understand now. 

The smart tab works based on selected cells as you have the ability to create repeating sequences or numbered sequences by selecting multiple frames and dragging right. For example: A2,A1 dragged out would create A2,A1,A2,A1,etc...  A1,A2 dragged out would create A1,A2,A3,A4,etc.  As you extend right, the new cells become part of the selected cells which is why you can immediately back it up and can only back up to the 1st selected cell.  I don't think this should change.

A faster way to preform what you show would be to click on the cell dragbar that connects all the exposed cells next to each other,.  This automatically selects them so you can then drag the last cell back to reduce the range.  Howerver, if I knew I wanted to remove a certain number of cells like that, i think it would be more efficient for me to just select the cells and then hit Delete instead of trying to drag it back.

**Comment by ladlon** [comment]
Could you show me a vid of what you are describing with your 'A faster way to perform what you show'?  I'm not sure I'm following what you are describing.

Are you talking about the bar going along the top of the cells (in the timeline)?

What I find (very often) that I am wishing I could do (and, out of muscle memory and (false) assumption), is simply grab a (drawing) key, and move it earlier or later (while keeping everything around it intact, in cases where there are more occupied cells after it).

In cases where I don't have occupied cells after it (...usually just a random amount of hold/exposure, as I work linearly through the lipsync), the shifting of the cells following the edit area is kind of irrelevant and harmless, as it's just 'temporary tail'.

Using the smart tab as I have been up to now, I've had to select the cells before the edit area, as seen in the video.... usually some random amount that only requires I select at LEAST the number of frames I want to shift by.

So, I get the impression, as I have with a few things, that there is actually a better/alternate way to do this... but I'm still kind of searching for that.

In the case of nothing afterwards, ya, you could do a select and delete or cut...

In the case of occupied cells (with keys) after the edit area, I'm still searching for an ideal method.

So far, it seems I can select some frames left of the key I want to make earlier, do a CUT (so the gap gets closed, but then I have to compensate for the shift of the other keys by ADDING frames to the RIGHT of the target key.  Again, my instinct simply wants me to be able to just grab the key and SHIFT it over, without disturbing anything.

It seemed like I found it previously, in something like a shift-drag of a selected key... but for some reason, I either abandoned it or forgot(!) it.  Either way, it seems somewhat clunky and unintuitive... at least based on my expectations.

If there's some way to simply select and (drawing) keyframe and shift it left or right, without shifting/affecting any other keys, please let me know!

I know that there's a nice 'shift selected TRANSLATIONAL keys' feature already present... I just wish a similar thing was available for DRAWING keys.

**Comment by ladlon** [comment]
I guess, to keep this thread as the intended 'feature request' I should clarify that I'm simply hoping there would be some very simple 'grab and drag' method of shift (drawing) keys around, so you could quickly and easily tweek the timings of the keys.

**Comment by manongjohn** [comment]
Yes. I was talking about the bar at the top of the cells. It's the cell dragbar that selects all contiguous cells

I only say its a little faster in that you aren't dragging 2x, once to select and then again to shrink.

![shrink_range](https://user-images.githubusercontent.com/19245851/97524589-6ddffe00-197b-11eb-8ef5-adc6ca38a8c7.gif)

For most new users, moving drawings around/retiming has always been a bit of a learning curve and/or a source of issues in the OT world, I think.  For now, the best ways to do that are probably better discussed on the forum or on a discord with others who might offer more efficient ways to do that.

**Comment by ladlon** [comment]
Ah!  I see what you are doing... You're basically using that bar to do the 'select some frames before the edit point' part of my procedure, allowing you to then move the handle to the left.  Neat trick.

That said... I still feel it would be more intuitive and consistant if you had this (or SOME) tool where you could simply drag forward or backward, rather than requiring a different method for one direction, vs another... Plus, this is one more step compared to what I'm suggesting.. and I can see on this clip  that you TOO were having a LITTLE bit of 'trouble' positioning the mouse pointer to the proper area on the bar before you could click.  It's a tiny micro-pause... but cumulatively, it adds to a lot of wasted time (and effort).  It's like pulldown menus... Sure, they work, and are perfectly acceptable... except when dealing with very repetitive tasks, and then their efficiency (having to move the mouse pointer, click on the menu, and then apply the micro-coordination required to find and position the mouse pointer over the appropriate menu item) becomes an issue... especially when compared to some alternative like, say, a keyboar shortcut.

It's nitpicky, to be sure... but when we are talking about hours of repeated actions, even things like the distance you have to move the mouse to get to something (or the fact that you have to actually relocate the mouse at all, rather than hitting a key) does add an arguably significant 'busywork' and requirement for 'precision' on the user (...in the case of something like having to position the mouse over a relatively small target area and click, vs being able to rapidly hit a keyboard key, keeping the mouse where it is needed, and not requiring pauses while you 'aim the mouse' over small target areas.

Just like a badly designed mouse, long sessions of it can create 'stress' (for lack of better word) on the user... and certainly make workflow slower or less instinctive/flowing that it CAN be.

I hope that doesn't come off as lofty and arogant....  I'm no expert in that sort of thing, but I do see 'inefficiencies' and areas that can be simplified and streamlined.  That sort of optimization can make a piece of software really shine, and cause people to really praise it's fast, intuitive workflow (and resulting high productivity!).

**Comment by manongjohn** [comment]
The pause/slowness you are seeing is because I was using a trackpad with my non-dominant hand to do this, which I don't normally use when I use the software.  I'm usually pretty good at targeting the bar otherwise. 

However, I'll concede there are inefficiencies with the software and leave it at that.

**Comment by ladlon** [comment]
Fair enough.  Thanks for the interesting discussion.  Cheers!
